src/modules/dashboard/components/AdvancedDialog/AdvancedDialog.tsx

```diff
@@ -104,6 +105,8 @@ const AdvancedDialog = ({
   const canEnterRecovery =
     hasRegisteredProfile && canEnterRecoveryMode(allUserRoles);
 
+  const canUpgradeVersion = hasRegisteredProfile && hasRoot(allUserRoles);
```

Comment:
you don't need to add this variable, you can just you can just use `hasRootPermission` from line 103 - it's the same value (same root permission that is being used for several parts)